crates/services/src/seqlock.rs

```diff
@@ -0,0 +1,158 @@
+//! A simple implementation of a sequential lock.
+//! More details: <https://docs.kernel.org/locking/seqlock.html>
+
+use std::{
+    cell::UnsafeCell,
+    sync::atomic::{
+        AtomicU64,
+        Ordering,
+    },
+};
+
+/// A simple implementation of a sequential lock.
+/// some usage of unsafe, T must be Copy
+#[derive(Debug)]
+pub struct SeqLock<T> {
```

Comment:
I don't think this struct needs to be public. Instead I'd propose just exposing a public function to create the reader and writer, similar to how channels are created.

```suggestion
struct SeqLock<T> {
```

Reply:
needs to be public because the constructor returns the writer and reader